controller.py

```diff
@@ -272,33 +276,45 @@ def libraries_opds(self, live=True, location=None):
             joinedload('hyperlinks', 'resource'),
             joinedload('hyperlinks', 'resource', 'validation'),
         )
+        alphabetical = alphabetical.options(defer('logo'))
```

Comment:
This stops the database from pulling the `logo` field into memory. The presence of this huge field in memory while doing joins is apparently what causes the slowdown.